abi-versions/vNEXT/README.md

```diff
@@ -0,0 +1,919 @@
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Get up to max_size bytes from the `buffer_type`, starting from `offset`. Bytes are written into
+buffer slice (`return_buffer_data`, `return_buffer_size`), and buffer flags are written into
+`return_flags`.
+
+
+### `proxy_set_buffer`
+
+* params:
+  - `i32 (proxy_buffer_type_t) buffer_type`
+  - `i32 (offset_t) offset`
+  - `i32 (size_t) size`
+  - `i32 (const char*) buffer_data`
+  - `i32 (size_t) buffer_size`
+  - `i32 (uint32_t) flags`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Set content of the buffer `buffer_type` to the bytes (`buffer_data`, `buffer_size`), replacing
+`size` bytes, starting at `offset` in the existing buffer.
+
+
+### `proxy_get_map`
+
+* params:
+  - `i32 (proxy_map_type_t) map_type`
+  - `i32 (const char**) return_map_data`
+  - `i32 (size_t*) return_map_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Get all key-value pairs from a given map (`map_type`).
+
+
+### `proxy_set_map`
+
+* params:
+  - `i32 (proxy_map_type_t) map_type`
+  - `i32 (const char*) map_data`
+  - `i32 (size_t) map_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Set all key-value pairs in a given map (`map_type`).
+
+
+### `proxy_get_map_value`
+
+* params:
+  - `i32 (proxy_map_type_t) map_type`
+  - `i32 (const char*) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const char**) return_value_data`
+  - `i32 (size_t*) return_value_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Get content of key (`key_data`, `key_size`) from a given map (`map_type`).
+
+
+### `proxy_set_map_value`
+
+* params:
+  - `i32 (proxy_map_type_t) map_type`
+  - `i32 (const char*) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const char*) value_data`
+  - `i32 (size_t) value_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Set or replace the content of key (`key_data`, `key_size`) to the value (`value_data`, `value_size`)
+in a given map (`map_type`).
+
+### `proxy_add_map_value`
+
+* params:
+  - `i32 (proxy_map_type_t) map_type`
+  - `i32 (const char*) key_data`
+  - `i32 (size_t) key_size`
+  - `i32 (const char*) value_data`
+  - `i32 (size_t) value_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Add key (`key_data`, `key_size`) with the value (`value_data`, `value_size`) to a given map
+(`map_type`).
+
+
+### `proxy_remove_map_value`
+
+* params:
+  - `i32 (proxy_map_type_t) map_type`
+  - `i32 (const char*) key_data`
+  - `i32 (size_t) key_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Remove key (`key_data`, `key_size`) from a given map (`map_type`).
+
+
+### `proxy_get_property`
+
+* params:
+  - `i32 (const char*) property_path_data`
+  - `i32 (size_t) property_path_size`
+  - `i32 (const char**) return_property_value_data`
+  - `i32 (size_t*) return_property_value_size`
+* returns:
+  - `i32 (proxy_result_t) call_result`
+
+Get property (`property_path_data`, `property_path_size`).
```

Comment:
are there a standard set of properties expected to be supported by every host?
the current root id comes to mind